### ISSUE TITLE
[wip] Serverside/API work for People, Followers, Following lists

### ISSUE THREAD

**Issue by xekc** [Lokiedu/libertysoil-site#939]
## Current

Currently in profile settings there's a "Followers" page that we with this work will remove: 

![image](https://user-images.githubusercontent.com/1575753/32105400-370c97e2-bb20-11e7-8d7e-03b5578e18c2.png)

Instead we'll build a full site users directory "People" (we have started moving in that direction, but abandoned it later). 

User will open "People" and then will click on a button to see his friends (contacts) as a sub-set from all users of the site. User will be able to open his friends list directly too.

## Concept design

All people list available as a sidebar interface overlay or via separate URL:

![2017-my3yr](https://user-images.githubusercontent.com/1575753/32108410-7e6a37da-bb29-11e7-899c-6db732dbb527.png)

List by default is sorted alphabetically. Other sorting options will be possible.

Clicking a button under site logo same interface switches to "Show my contacts only" mode:

![image](https://user-images.githubusercontent.com/1575753/32108550-f8868028-bb29-11e7-95e8-55939a838c5a.png)

Further sub-sets in that interface are possible, such as "Following me":

![image](https://user-images.githubusercontent.com/1575753/32108580-0de9c1e6-bb2a-11e7-9ca0-68fa47f6241e.png)

"I follow": 
![image](https://user-images.githubusercontent.com/1575753/32108589-18784556-bb2a-11e7-9794-26d13f852f88.png)

"I have blocked:"
![image](https://user-images.githubusercontent.com/1575753/32108604-22b4e65a-bb2a-11e7-8d63-852024094d8d.png)

## Implementation

This issue is dedicated to data / server side / API work only and is fully with @voidxnull and @indeyets 

- [ ] What needs to be developed to deliver full lists of user in an alphabetical order to the client side? Is everything ready?

  - [ ] We expect paging-by-letter here. When the whole list is visible - first (alphabetically) 10 are displayed:

![2017-uijls](https://user-images.githubusercontent.com/1575753/32109170-cf04ee36-bb2b-11e7-8498-6ada14605f34.png)

..the rest are hidden under "more" that opens a separate interface:

![2017-y6g0t](https://user-images.githubusercontent.com/1575753/32109260-1cfdb01e-bb2c-11e7-9eab-180535233d99.png)

where the full list of users is available with simple next/previous paging.

**What needs to be changed in API to make this possible?**

We want to build this interface with all other languages in mind. **Are there any implications in other alphabets on the data/API level?** Currently it's out of scope but we'll want to do French / German very soon.

  - [ ] What needs to be done to deliver the number of posts from every user through API (if it's not implemented yet)?

  - [ ] We will use both auto-load of the contacts list - and paging. What needs to be done to make both work? 

### Followers

Old https://alpha.libertysoil.org/settings/followers hyperlink works and opens this interface in sidebar: 

![image](https://user-images.githubusercontent.com/1575753/32109503-d575b06a-bb2c-11e7-874f-e07314b69af1.png)

- [ ] Accurate number of users following me is displayed
- [ ] Users are displayed sorted alphabetically
- [ ] Each letter is a sub-header, empty (no contacts currently) letters not shown.
- [ ] Each letter contains up to 30 users sorted alphabetically with "x more" link displayed afterwards.

- [ ] Checkbox icon is displayed for a user that is following me. 
  - [ ] Checkbox icon is clickable. Clicking changes it to a "Loading" indicator then to a "Add" icon, (means you have unfollowed this person). 

- [ ] "Add" icon is displayed for users that follow me, but I'm not following. 
  - [ ] Clicking "Add" icon changes it to a "Loading" indicator, then to a "Checkbox" icon (means you also now follow this user).

**Following (I follow) & Blocked**

- [ ] Implementation is the same but there's no follow/unfollow checkbox. To follow someone you have to do it from their profile block, not here.
- [ ] Implementation of the Blocked users list hasn't started yet, but I think it's possible to consider this as an exclusive group that you can place any user of the site. Further consequences of being in this group are out of scope here, so the only outcome of placing someone in the group "Blocked" will be seeing him on that list.
- [ ] Clicking "Blocked" icon icon unblocks the user and removes him from the list.
